### ISSUE TITLE
Stop Reports from flickering when opening tab

### ISSUE THREAD

**Issue by amulware** [toggl/mobileapp#4879]
When I open the reports tab on Daneel the report for the first time, it remains empty at first and then the report fades in. When I navigate somewhere else and back it displayed flickers and seems to reload.

Several problems with that:

- Most importantly, it should definitely not flicker. It should always smoothly animate (fade) between states.
- Secondly, we have an empty/loading view which should be showing when it's loading the report instead of an empty screen
- Thirdly, it should not reload the same report when I navigate back and forth.

On the last point, I propose to only reload if the view has not been seen in at least five minutes (this state can be kept in the view so that if it gets cleaned up for whatever reason it'll load again next time you open it).

I thought this was already part of the original designs, but it was either not communicated properly, or it was implemented wrongly.

**Comment by amulware** [comment]
Added `foundation` label since the reloading logic should probably be shared code.

**Comment by maritoggl** [comment]
Do I understand correctly that what you propose also means that if I change the report timeframe at first, then navigate back and forth, the report timeframe I initially selected stays the same (unless five minutes go by and the report reloads)?

**Comment by amulware** [comment]
Yes. I didn't realise it was changing back to the default, I think that's quite bad since going back and forth between tabs is so easy. The calendar also reset scroll-state btw, which we might wanna fix. The log doesn't, it remembers where you were, which I believe is the expected behaviour.

**Comment by amulware** [comment]
After testing it, Android doesn't actually behave like this - it keeps tab state intact as expected.

**Comment by maritoggl** [comment]
Sounds reasonable to me, I have no objections.

**Comment by amulware** [comment]
Note to the dev taking this: If your solution involves saving tab state in such a way that the calendar view also remembers its scroll position, that's great! If it does not, please **create an issue to make the calendar remember scroll position when switching between tabs**.

**Comment by arturs105** [comment]
Unassigning myself from this issue since i'm gonna work on investigating unrepresentable date error. Here's what i managed to find out:
* Last good commit (loading views work): `b93b0bd09762d9b309a053fade8caecc53e0547b `
* First bad commit (loading views do not work anymore): 047b26b444dbffcf1167d22ee73b438d42fdf5bb

I also found the reason, why the loading views aren't working - the table view is waiting for data to arrive from the view model and only shows _any_ cells (including loading views) when the data arrives. To fix this you need to `StartWith` and empty chart. Change this line:
```
SegmentsObservable = segmentsSubject.CombineLatest(DurationFormatObservable, applyDurationFormat);
```
To this:
```
SegmentsObservable = segmentsSubject.CombineLatest(DurationFormatObservable, applyDurationFormat).StartWith(new ChartSegment[0]);
```

**Comment by dshr** [comment]
Shall I add a way to manually refresh the reports view? Since right now the only way to do that, apart from reopening the tab, is the re-select the date from the calendar, which is rather fiddly imo. A pull-to-refresh should work okay

**Comment by dshr** [comment]
Unassigning myself and moving this to 2.0 since it's going to be solved with MVExit.

**Comment by juanlaube** [comment]
The empty state shows while the report is loading was fixed and will be released in Daneel 2.0. I'm moving this issue to the Daneel 2.1 release to continue with the flickering and reloading when switching tabs.

**Comment by dshr** [comment]
Unassigning myself since I'm on vacation and other people may try fixing this.

**Comment by dshr** [comment]
Closed by #5559 5559
